src/v1/controllers/TenantUpdate/updateTenant.ts

```diff
@@ -0,0 +1,84 @@
+import { Request, Response } from 'express';
+import * as _ from 'lodash';
+import * as uuid from 'uuid';
+import httpStatus from 'http-status';
+import { getTenantById, updatetenant } from '../../services/tenantService';
+import { schemaValidation } from '../../services/validationService';
+import logger from '../../utils/logger';
+import { errorResponse, successResponse } from '../../utils/response';
+import tenantUpdateJson from './updateTenatValidationSchema.json';
+import { getTenantBoardById, updatetenantBoard } from '../../services/tenantBoardService';
+
+export const apiId = 'api.tenant.udpate';
+
+type UpdateFunction = (data: any, id: number, tenant_id: number) => Promise<any>;
+
+type getFunction = (id: any) => Promise<any>;
+
+//update action for tenant and tenant board
+const updateActions: Record<string, UpdateFunction> = {
+  tenant: async (data, id) => await updatetenant(data, id),
+  tenantBoard: async (data, id, tenant_id) => await updatetenantBoard(data, id, tenant_id),
+};
+
+//get action for tenant and tenant board
+const getActions: Record<string, getFunction> = {
+  tenant: async (id) => await getTenantById(id),
+  tenantBoard: async (id) => await getTenantBoardById(id),
+};
+
+export const tenantUpdate = async (req: Request, res: Response) => {
+  const requestBody = req.body;
+  const id = uuid.v4();
+  const key = _.get(req, 'params.key');
+
+  try {
+    //validating the update schema
+    const isRequestValid: Record<string, any> = schemaValidation(requestBody, tenantUpdateJson);
+    if (!isRequestValid.isValid) {
+      const code = 'TENANT_INVALID_INPUT';
+      logger.error({ code, apiId, requestBody, message: isRequestValid.message });
+      return res.status(httpStatus.BAD_REQUEST).json(errorResponse(id, httpStatus.BAD_REQUEST, isRequestValid.message, code));
+    }
+
+    //validating the tenat is already exist
+    const isTenantExists = await checkTenantExists(_.get(requestBody, ['id']), key);
+    if (!isTenantExists) {
+      const code = 'TENANT_NOT_EXISTS';
+      logger.error({ code, apiId, requestBody, message: `${key} not exists with id:${_.get(requestBody, ['id'])}` });
+      return res.status(httpStatus.CONFLICT).json(errorResponse(id, httpStatus.CONFLICT, `${key} not exists with id:${_.get(requestBody, ['id'])}`, code));
+    }
+
+    //update existing tenant or tenant board
+    const updateFunction: UpdateFunction = updateActions[key];
+    const keysToOmit = _.has(requestBody, 'tenant_id') ? ['id', 'tenant_id'] : ['id'];
+    const updateData = _.omit(requestBody, keysToOmit);
+    const updateId = _.get(requestBody, ['id']);
+    const updateTenantId = _.get(requestBody, ['tenant_id'], '');
+
+    await updateFunction(updateData, updateId, updateTenantId);
```

Comment:
how are we managing to update tenant board fields with this?

Reply:
i have created a predefined object  based on the value from params it will access the object key to call the function ,it sort similar to hash map but not full obtained by it .